### PR TITLE
Optional buffer format and docs

### PR DESCRIPTION
**Pull request by delta003**

### CODE REVIEW

File: pydust/src/types/buffer.zig [hunk 1]
```diff
@@ -57,7 +57,9 @@ pub const PyBuffer = extern struct {
     // If ndim == 0, the memory location pointed to by buf is interpreted as a scalar of size itemsize.
     // In that case, both shape and strides are NULL.
     ndim: c_int,
-    format: [*:0]const u8,
+    // A NUL terminated string in struct module style syntax describing the contents of a single item.
```
Comment (robert3005) [inline pydust/src/types/buffer.zig]:
```suggestion
    // A NULL terminated string in struct module style syntax describing the contents of a single item.
```

File: docs/guide/_6_buffers.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Python objects implementing the [Buffer Protocol](https://docs.python.org/3/c-api/buffer.html#) can be used with zero copy.
+
+```zig
+--8<-- "example/buffers.zig:sum"
+```
+
+This function accepts Python [arrays](https://docs.python.org/3/library/array.html#module-array), Numpy arrays, or any other buffer protocol implementation.
+
+```python
+import numpy as np
```
Comment (gatesn) [inline docs/guide/_6_buffers.md]: Would be good to import this from test_buffers.py

File: mkdocs.yml [hunk 3]
```diff
@@ -25,6 +25,7 @@ nav:
     - 'Exceptions': "guide/exceptions.md"
     - 'Testing': "guide/_4_testing.md"
     - 'Memory Management': "guide/_5_memory.md"
+    - 'Buffer Protocol': "guide/_6_buffers.md"
```
Comment (gatesn) [inline mkdocs.yml:28]: I've given up on numeric prefixes, they're kind of annoying

File: docs/guide/_6_buffers.md (hunk 2 above)
Comment (delta003) [reply docs/guide/_6_buffers.md]: Done

**Review by robert3005 — APPROVED** [review]

**Pull request merged**